src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -1459,15 +1469,42 @@ std::pair<int64_t, int64_t> program::get_estimated_device_mem_usage() {
               [](program_node* const& lhs, program_node* const& rhs) {
                   return (lhs->get_output_layout().bytes_count() > rhs->get_output_layout().bytes_count());
               });
-
+    auto& engine = get_engine();
+    int64_t host_alloc = 0;
+    GPU_DEBUG_GET_INSTANCE(debug_config);
     // just to prevent the memories from being freed during allocation
     std::unordered_set<memory::ptr> allocated_mem_ptrs;
     for (const auto& node : nodes_to_allocate) {
         auto out_size = node->get_output_layout().bytes_count();
         if (out_size > max_alloc_size) {
             // to consider: if the base batch size is > 1, should we allow this single output allocation to host?
-            continue; // to be allocated to host
+            host_alloc += out_size;
+            continue;
         }
+        #ifdef __unix__
+        // Check whether the host mem allocation might exceed avialalbe system VRAM
+        if (engine.get_device_info().dev_type == cldnn::device_type::integrated_gpu) {
```

Comment:
As both branches are almost the same, could it be simplified to something as follows?
```cpp
auto total_host_alloc_size = out_size + host_alloc + engine.get_used_device_memory(allocation_type::usm_host);
if (engine.get_device_info().dev_type == cldnn::device_type::integrated_gpu) {
    total_host_alloc_size += engine.get_used_device_memory(allocation_type::usm_device);
}

if (total_host_alloc_size > cur_vmem) {
    GPU_DEBUG_IF(debug_config->verbose >= 1) {
        GPU_DEBUG_COUT << "Estimated mem usage calculated with default base batch size(16) exceeds the available virtual memory ("
                       << cur_vmem << ")" << std::endl;
    }
    return {-1L, -1L};
}
```

Reply:
I updated, thanks for the review.